lib/http/tracer.rb

```diff
@@ -45,16 +45,27 @@ def request(verb, uri, opts = {})
             options = HTTP::Options.new.merge(opts)
             parsed_uri = uri.is_a?(String) ? URI(uri) : uri
 
-            if ::HTTP::Tracer.ignore_request.call(verb, uri, options)
+            begin
+              skip = ::HTTP::Tracer.ignore_request.call(verb, uri, options)
+            rescue StandardError
```

Comment:
I think if the error is not an `Invalid URI` error, you probably want to notify about this exception if possible so that you can investigate further why this is rescuing - i.e. are there some exceptions that we don't want to catch, or rather might need to be mitigated in other ways?

Reply:
Yeah, in general just rescuing `StandardError` and ignoring it could cause other issues to go unnoticed. 

What is actually the original problem? Is there a error/stacktrace for it?

Reply:
this callback can have arbitrary code, it's not possible to know what type of errors could be triggered.  In the event that it blows up you have two options 

a) let your tracing code crash whatever the untraced code was trying to do 
b) treat all requests that are not explicitly allowed as if they should not be traced and have the untraced code continue on its way

I think adding logging in this event would be a good idea but there is no way to pass a logger into the instrumentation yet.  I'll likely add that in an upcoming version

Reply:
I'll backout the error handling for this release since silently failing without logging also stinks in that situation

Reply:
https://github.com/benedictfischer09/ruby-httprb-instrumentation/pull/3